__tests__/endpoints.test.js

```diff
@@ -126,3 +126,38 @@ describe('GET /api/articles', () => {
         })
     })
 })
+describe('GET /api/articles/:article_id/comments', () => {
+    describe('main functionality', () => {
+        test('responds with status 200 and an array of objects', () => {
+            return request(app)
+                .get('/api/articles/9/comments')
+                .expect(200)
+                .then((res) => {
+                    const comments = res.body.comments;
+                    expect(Array.isArray(comments)).toBe(true);
+                    comments.forEach((comment) => {
+                        expect(Array.isArray(comment)).toBe(false);
+                        expect(typeof comment).toBe('object');
+                    })
+                })
```

Comment:
Great error handling - but this main functionality test could be improved

Right now you're just checking that comments are objects, which is good, but you can be more specific! What about it's properties, and the datatype / value of those properties?

We can use the testing pattern toMatchObject in the following way for this example:

comments.forEach((comment) => {
                        expect(comment).toMatchObject({
                        // now we can define expected properties in here, either with datatypes, or exact values!
                        username: "Drodinger"
                        votes : 0
                        created_at: expect.any(String) // <- very useful testing pattern to know!
                        })
                    })
 Have a go at using this to write a really strong test here!